File: src/wallet_app.py
from src.model.history import History
from src.model.operation_type import OperationType
from src.model.type.result import Result
from src.model.wallet import Wallet
from src.service.wallet_service import print_wallet, deposit, withdrawal
from src.view.bar_view import bar
from src.view.history_view import print_history
from src.view.menu_view import print_menu


def run():
    wallet: Wallet = Wallet()
    history_list: list[History] = []

    print("Welcome")

    condition_running: bool = True
    while condition_running:
        print_menu("Status", "Deposit", "Withdraw", "History", "Exit")
        input_str: str = input("Choice: ")

        choice: int = int(input_str)
        if choice == 1:
            bar("Status")
            print_wallet(wallet)
            history_list.append(History(OperationType.STATUS, wallet))
        elif choice == 2:
            bar("Deposit")
            res: Result = deposit(wallet)
            if res.is_success:
                print(res.message)
                history_list.append(History(OperationType.DEPOSIT, wallet))
            else:
                print(res.message)
        elif choice == 3:
            bar("Withdraw")
            res: Result = withdrawal(wallet)
            if res.is_success:
                print(res.message)
                history_list.append(History(OperationType.WITHDRAW, wallet))
            else:
                print(res.message)
        elif choice == 4:
            print_history(history_list)
        elif choice == 5:
            condition_running = False
            print("Bye")
        else:
            print("RETRY")

File: lib/wallet_app.rb
require_relative 'model/history'
require_relative 'model/operation_type'
require_relative 'model/types/result'
require_relative 'model/wallet'
require_relative 'service/wallet_service'
require_relative 'view/bar_view'
require_relative 'view/history_view'
require_relative 'view/menu_view'

def run
  wallet = Wallet.new
  history_list = []

  puts 'Welcome'

  condition_running = true
  while condition_running
    print_menu('Status', 'Deposit', 'Withdraw', 'History', 'Exit')
    input_str = gets.chomp

    choice = input_str.to_i
    case choice
    when 1
      bar('Status')
      print_wallet(wallet)
      history_list << History.new(OperationType::STATUS, wallet)
    when 2
      bar('Deposit')
      res = deposit(wallet)
      if res.is_success
        puts res.message
        history_list << History.new(OperationType::DEPOSIT, wallet)
      else
        puts res.message
      end
    when 3
      bar('Withdraw')
      res = withdrawal(wallet)
      if res.is_success
        puts res.message
        history_list << History.new(OperationType::WITHDRAW, wallet)
      else
        puts res.message
      end
    when 4
      print_history(history_list)
    when 5
      condition_running = false
      puts 'Bye'
    else
      puts 'RETRY'
    end
  end
end
